src/commonMain/kotlin/com/algolia/search/configuration/AuthenticationImpl.kt

```diff
@@ -0,0 +1,10 @@
+package com.algolia.search.configuration
+
+import com.algolia.search.model.APIKey
+import com.algolia.search.model.ApplicationID
+
+
+internal data class AuthenticationImpl(
```

Comment:
Not sure there is a need for interface/impl: It would be easier for our users to simply have a `Credentials` data object, as I see no use-case for a custom `CredentialsImpl`